src/lock.rs

```diff
@@ -38,16 +38,25 @@ impl FromStr for Lock {
     type Err = Report;
 
     fn from_str(value: &str) -> Result<Self> {
-        let lock = match value {
-            "AccessShareLock" => Lock::AccessShareLock,
-            "RowShareLock" => Lock::RowShareLock,
-            "RowExclusiveLock" => Lock::RowExclusiveLock,
-            "ShareUpdateExclusiveLock" => Lock::ShareUpdateExclusiveLock,
-            "ShareLock" => Lock::ShareLock,
-            "ShareRowExclusiveLock" => Lock::ShareRowExclusiveLock,
-            "ExclusiveLock" => Lock::ExclusiveLock,
-            "AccessExclusiveLock" => Lock::AccessExclusiveLock,
-            _ => return Err(eyre!("invalid lock type {value}")),
+        let normalised = value.to_lowercase()
+            .replace(" ", "");
+
+        let normalised = if !normalised.ends_with("lock") {
+            normalised + "lock"
+        } else {
+            normalised
+        };
```

Comment:
Instead of normalising the string value (and thus allocating some memory in various places, having weird edge cases where we need to add the `lock` suffix), it's probably easier to just add more variants to the `match` statement 😅 

We're already using this bit of code for transforming the underlying SQL return values into the enum so this would mean we'd need to normalise them instead of mapping directly.

The [underlying code](https://godbolt.org/z/vza7oj3Kh) generation in Assembly looks mostly the same with more variants added (it's just a jump tree from what I can see, letter by letter to find the right variant) so will be pretty quick. `to_lowercase` and `replace` will require heap allocations and are a bit harder to think about the flow of the code.

Reply:
I can make that change, but it does make it a little less forgiving, for example these variations would continue to fail.

```
"access exclusive"
"ACCESS EXCLUSIVE LOCK",
"ACCESS EXCLUSIVE  "
```

Reply:
Yeah, we're somewhat bound by what Postgres is using internally here, since we need to be able to parse from the underlying database types and it would be nice if they match up well. `clippy` complains about the `Lock` enum having variants that end in `Lock`, but I think it would be a bit more work to tidy that up but still have the `FromSql` implementation work.

Perhaps we can keep the `match` the same as it is in `master` and then use the `_ =>` case to handle attempts to mangle the input into a more standardised variation (returning the same error if it doesn't find a match). Then we could handle all the cases you've just listed there but off the path of `FromSql`.

Reply:
I've pushed an update, though I'm not sure if I've followed exactly what you had in mind? I don't know rust so it might not be very idiomatic.